odin-dynamic/src/main/scala/com/permutive/logging/dynamic/odin/DynamicOdinLogger.scala

```diff
@@ -71,12 +71,27 @@ object DynamicOdinConsoleLogger {
 
   case class RuntimeConfig(
       minLevel: Level,
-      levelMapping: Map[String, Level] = Map.empty
+      levelMapping: Map[String, LevelConfig] = Map.empty
```

Comment:
Technically this _should_ have failed the bincompat checks 🤔 as the type has changed, I'll try and work out why that isn't working.

In the meantime, I'm trying to think of the best way to make a bincompat change here and I think you might need to create a new case class that uses the new level config. Then have a smart constructor that takes this old style config and translates it to the new.

Reply:
Ah, yes that's a good point. I'll see what I can do, but the bincompat doesn't fail locally for me either, which doesn't make me too confident that changes I make are actually ok.

Could it be that, because it's a type parameter it _is_ technically bincompat because of erasure? That wouldn't seem right to be but....

Reply:
> Could it be that, because it's a type parameter it _is_ technically bincompat because of erasure?

💯 that's how the JVM works. It won't save your from runtime errors, but it won't be a _linking_ error.

Reply:
Just looking at the [bin compat guide](https://docs.scala-lang.org/overviews/core/binary-compatibility-for-library-authors.html#changing-a-case-class-definition-in-a-backwards-compatible-manner). Would this maintain bin compat?  
```scala
case class RuntimeConfig private (
      minLevel: Level,
      levelMapping: Map[String, Level] = Map.empty,
      disabledClasses: List[String] = Nil
) {
  def withDisabledClasses(disabled: List[String]) = ???
}
object RuntimeConfig {
    def apply(minLevel: Level, levelMapping: Map[String, Level] = Map.empty) = RuntimeConfig(minLevel, levelMapping)
}
```